deepchem/models/wandblogger.py

```diff
@@ -0,0 +1,144 @@
+import os
+import math
+import logging
+import importlib.util
+from typing import Any, Callable, Dict, Iterable, List, Optional, Sequence, Tuple, Union
+from deepchem.data import Dataset
+from deepchem.metrics import Metric
+from deepchem.models.callbacks import ValidationCallback
+
+logger = logging.getLogger(__name__)
+
+def is_wandb_available():
+    # any value of WANDB_DISABLED disables wandb
+    if os.getenv("WANDB_DISABLED"):
+        logger.warning(
+            "Using the `WAND_DISABLED` environment variable is deprecated and will be removed in v5. Use the "
+            "--report_to flag to control the integrations used for logging result (for instance --report_to none)."
+        )
+        return False
+    return importlib.util.find_spec("wandb") is not None
+
+class WandbLogger(object):
+    """
+    Weights & Biases Logger
+    """
+
+    def __init__(self,
+                 metrics: List[Metric],
+                 train_dataset: Dataset,
+                 eval_dataset: Optional[Dataset] = None,
+                 eval_logging_strategy: Optional[str] = "step",
+                 name: Optional[str] = None,
+                 entity: Optional[str] = None,
+                 project: Optional[str] = None,
+                 save_dir: Optional[str] = None,
+                 mode: Optional[str] = "online",
+                 id: Optional[str] = None,
+                 resume: Optional[Union[bool, str]] = None,
+                 anonymous: Optional[str] = "never",
+                 log_model: Optional[bool] = False,
+                 **kwargs):
+
+        has_wandb = is_wandb_available()
+        assert has_wandb, "WandbLogger requires wandb to be installed. Run `pip install wandb --upgrade`."
+        if has_wandb:
+            import wandb
+            self.wandb = wandb
```

Comment:
Huggingface's setup is similar to the one currently used. Should we follow their format or revert back to the old format where we initialize the run using `self.wandb_run = wandb.init(...)`

Reply:
Will make the changes. However I'll keep the import part and change it to
`import wandb
 self._wandb = wandb`
since we need a wandb object to call wandb.init(...).
Just doing `import wandb` throws an error since the wandb.init(...) can't access wandb inside the setup function